S/SHTns/build_tarballs.jl

```diff
@@ -5,11 +5,13 @@ using BinaryBuilder, Pkg
 include(joinpath(@__DIR__, "..", "..", "platforms", "microarchitectures.jl"))
 
 name = "SHTns"
-version = v"3.5.2"
+version = v"3.6.6"
+sha256sum = "f060757ed6914c837cc2b251d370078e4c92b6894fef7aac189a9a1f5f1521a2"
+url = "https://gricad-gitlab.univ-grenoble-alpes.fr/schaeffn/shtns/-/archive/v3.6.6/shtns-v$(version).tar.gz"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://bitbucket.org/nschaeff/shtns/downloads/shtns-$(version).tar.gz", "dc4ac08c09980e47c71d79d38696c5d1d631f86c2af1ce8aad5d21f7fd2c05b9")
+    ArchiveSource(url, sha256sum)
```

Comment:
not that I mind, but I wonder why this change? It is just a bit unusual -- which is irrelevant when you only look at this recipe, of course, but causes one to stop and wonder when one has to deal with dozens and hundreds of recipe files. From that perspective, following the principle of least surprise is quite helpful, i.e., don't deviate from the common pattern unless you have a good reason.


Alas, that's highly subjective, I wouldn't to hold up this PR over that :-)

Reply:
Thanks for taking the time to review. Since I very rarely deal with BinaryBuilder recipes, the personal reason for making this change is to remember that the control sum of the tarfile can be computed with the `sha256sum` Linux command. I would happily to revert to the usual form.